elastic/kibana
# [Reporting] Add new ILM policy notice to deprecations service

## Summary

Continuation of https://github.com/elastic/kibana/pull/100130 and fix https://github.com/elastic/kibana/issues/81544

This contribution adds an integration with the deprecation service. The deprecation flags the new reporting policy and provides a "quick resolution" in the upgrade assistant which uses the new "PUT <base-path>/deprecations" endpoint.

## How to review

1. Check the added copy for any obvious errors or wording issues (requires proper copy review)
2. Start Kibana & ES on a trial license
3. Add sample flight data
4. Choose the Global Flight dashboard and choose **Share** > **PDF Reports** > **Generate report**
5. Edit upgrade assistant to show deprecations UI. Open `x-pack/plugins/upgrade_assistant/common/constants.ts` and set:

```
export const UA_READONLY_MODE = false;
```
6. Go to the upgrade assistant and click on the "Quick resolve" button (screenshot 👇🏻)

![Screenshot 2021-05-26 at 14 19 46](https://user-images.githubusercontent.com/8155004/119658472-79ba7280-be2d-11eb-9ddc-033ff455722e.png)


### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
